lib/welcome/pages/forgot_password_page.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_app/util/email_field.dart';
+import 'package:flutter_app/util/loading_button.dart';
+import 'package:progress_state_button/progress_button.dart';
+import 'package:flutter_app/util/supabase.dart';
+
+class ForgotPasswordPage extends StatefulWidget {
+  final String? initialEmail;
+  const ForgotPasswordPage({super.key, this.initialEmail = ""});
+
+  @override
+  State<ForgotPasswordPage> createState() => _ForgotPasswordPageState();
+}
+
+class _ForgotPasswordPageState extends State<ForgotPasswordPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Reset password"),
+      ),
+      body: Center(
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 40, vertical: 20),
+            child: ForgotPasswordForm(
+              initialEmail: widget.initialEmail,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class ForgotPasswordForm extends StatefulWidget {
+  final String? initialEmail;
+  const ForgotPasswordForm({super.key, this.initialEmail = ""});
+
+  @override
+  State<ForgotPasswordForm> createState() => _ForgotPasswordFormState();
+}
+
+class _ForgotPasswordFormState extends State<ForgotPasswordForm> {
+  ButtonState _state = ButtonState.idle;
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  late final TextEditingController emailController;
+
+  @override
+  void initState() {
+    super.initState();
+    emailController = TextEditingController(text: widget.initialEmail);
+  }
+
+  void onSubmit() async {
+    if (_formKey.currentState!.validate()) {
+      setState(() {
+        _state = ButtonState.loading;
+      });
+
+      supabaseClient.auth.resetPasswordForEmail(emailController.text,
+          redirectTo: kIsWeb ? null : 'io.supabase.flutter://reset-callback/');
+
+      await Future.delayed(const Duration(seconds: 1));
+      setState(() {
+        _state = ButtonState.success;
+      });
+      await Future.delayed(const Duration(seconds: 2));
+
+      onMailSent();
```

Comment:
```suggestion
      if (mounted) {
        Navigator.of(context).pop();
      }
```

you can also check if widget is 'mounted', then you can use context in an async method.